src/test/java/org/programmers/springbootbasic/voucher/VoucherServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package org.programmers.springbootbasic.voucher;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.programmers.springbootbasic.voucher.model.FixedAmountVoucher;
+import org.programmers.springbootbasic.voucher.model.PercentDiscountVoucher;
+import org.programmers.springbootbasic.voucher.model.Voucher;
+import org.programmers.springbootbasic.voucher.model.VoucherType;
+import org.programmers.springbootbasic.voucher.repository.VoucherRepository;
+import org.programmers.springbootbasic.voucher.service.VoucherService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+class VoucherServiceTest {
+
+    @Mock
+    VoucherRepository voucherRepository = mock(VoucherRepository.class);
+
+    @InjectMocks
+    VoucherService voucherService = new VoucherService(voucherRepository);
+
+    @Test
+    @DisplayName("유효하지 않은 바우처 넘버로 바우처를 생성할 수 없다.")
+    void createVoucherByInvalidVoucherTypeTest(){
+        assertThatThrownBy(() -> VoucherType.findByNumber(3))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("잘못된 바우처 넘버입니다.");
+    }
+
+    @Test
+    @DisplayName("바우처를 생성할 수 있다.")
+    void createVoucherTest() {
+        VoucherType voucherType = VoucherType.FIXED;
+        long value = 100L;
+        var voucherCreatedByType = voucherType.create(value);
+
+        when(voucherRepository.insert(any())).thenReturn(voucherCreatedByType);
+
+        Voucher voucher = voucherService.createVoucher(voucherType, value);
+
+        assertThat(voucher.getVoucherId()).isEqualTo(voucherCreatedByType.getVoucherId());
+    }
+
+    @Test
+    @DisplayName("바우처 아이디로 원하는 바우처를 조회할 수 있다.")
+    void getVoucherByIdTest() {
+        Voucher voucher = new FixedAmountVoucher(UUID.randomUUID(), 3000L);
+        when(voucherRepository.findById(voucher.getVoucherId())).thenReturn(Optional.of(voucher));
+
+        Voucher retrievedVoucher = voucherService.getVoucher(voucher.getVoucherId());
+
+        assertThat(retrievedVoucher.getVoucherId()).isEqualTo(voucher.getVoucherId());
+        verify(voucherRepository).findById(voucher.getVoucherId());
+    }
+
+    @DisplayName("존재하지 않는 아이디로는 바우처를 조회할 수 없다.")
+    @Test
+    void getVoucherByNoIdTest() {
+        assertThatThrownBy(() -> voucherService.getVoucher(UUID.randomUUID()))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessageContaining("바우처를 찾을 수 없습니다");
+    }
```

Comment:
```suggestion
    void getVoucherByNoIdTest() {
	    UUID voucherId = UUID.randomUUID();
        assertThatThrownBy(() -> {
	        voucherService.getVoucher(voucherId);
        })
                .isInstanceOf(RuntimeException.class)
                .hasMessageContaining("바우처를 찾을 수 없습니다");
    }
```

큰 문제가 되는 코드는 아니지만 테스트 코드 작성 시에는 메서드 안에 또다른 메서드를 호출하는 방식으로 작성하는 것은 지양합니다.